### ISSUE TITLE
[question] Unread messages

### ISSUE THREAD

**Issue by RainLoop** [RainLoop/rainloop-webmail#611]
I have added a new opportunity.

![22](https://cloud.githubusercontent.com/assets/5530092/7356776/084ded38-ed33-11e4-9cfc-5bf3143411d1.gif)

**Reason:**

Many users have a very large number of folders in their accounts. This is not good, as RainLoop wants to check them all and this creates a large number of connections to the mail server. But this is not necessary, as the number of unread messages in the folder needs to know only for some specific folders.

What do you think about this?

**Comment by megadr01d** [comment]
1. It helps people with a large number of folders
2. It doesn't hurt people with a small number of folders
3. It's optional

I can't see any reason to dislike it.

**Comment by skibbipl** [comment]
Will selecting hide message count remove number badge from folder? Perhaps RL can cache current unread messages and refresh the count only when user enters particular folder? And for selected folders the badge should be updated always.

**Comment by RainLoop** [comment]
> Will selecting "hide message count" remove number badge from folder?

Yes, otherwise it may cause confusion of users.
In most cases, users do not understand what "requests to the server" is, but what is the number of messages they understand.

**Comment by RainLoop** [comment]
At the same time, system folders  (Inbox, Spam ... etc.) will always display the number of unread messages.

**Comment by skibbipl** [comment]
Not sure about removing the badge - I prefer to know how many unread items I have in different folders. On the other hand this causes multiple IMAP requests. Therefore I asked about a "smarter" solution with cache option: non important folders could refresh unread items only when selecting the folder - some folders are less busy than others so overall number of IMAP requests will be significantly lower. When refreshing the whole page, unread items for those folders could be taken from cache, preventing useless IMAP requests.
Anyway this is just a suggestion.

**Comment by jelle-e** [comment]
Great idea! Agree with @megadr01d 

Maybe call it "Show/hide number of new messages" or "Show/hide new message count" ?
It might be more clear for 'normal' users, but I'm not sure...

**Comment by RainLoop** [comment]
@skibbipl 

Yes, of course, there are many optimization.

> non important folders could refresh unread items only when selecting the folder

And it would be perceived as a bug. 
It is difficult to decide which folders are important. That's why I decided to add this option.

@jelle-e 

>  or "Show/hide new message count" ?

Yes, I like this version.


In fact it is not a big problem when you have a small number of folders.

**Comment by RainLoop** [comment]
@jelle-e 

And yes, if you have other suggestions about english translation (I'm not native)
I would glad to hear them.

**Comment by megadr01d** [comment]
I think `Show/hide message count` is enough. Message counts are for *unread* messages, not necessarily *new* messages (except for the Draft folder which is common rule to show total messages).

**Comment by ervee** [comment]
I'd make it ```Show/hide unread counter```. Or the bit longer ```Show/hide unread messages counter```.

**Comment by jelle-e** [comment]
@ervee agreed (long version for clarity)

**Comment by skibbipl** [comment]
@RainLoop
I mean that non important folders = folders for that I set hide unread messages. Maybe rephrase it - "do not refresh automaticaly number of new messages."

**Comment by megadr01d** [comment]
I think I get what skibbipl is saying. With that new folder option, we're not just **showing/hiding** the message count, we're actually **enabling/disabling** new message fetching for specific folders to avoid connection requests. So, maybe the new option label should be changed to something like, as skibbipl points out, **check/don't check for new messages**.

So, taking the following example

~~~
Inbox (1)
Drafts
Sent
Trash

--------

✅ Custom Folder 1 (3)
☑️ Custom Folder 2
✅ Custom Folder 3 (1)
~~~

Key:

✅ Fetch new messages automatically (fetch, retrieve, check, whatever suits you)
☑️ Don't fetch new messages automatically

manually selecting `Custom Folder 2` would fetch its messages and the message counter would update.

@skibbipl & @RainLoop did I get it right?

**Comment by skibbipl** [comment]
@megadr01d 
Yeah, this is my idea nicely written  :+1:

**Comment by RainLoop** [comment]
So mote it be! :)

**Comment by ervee** [comment]
Did this get implemented? I can't find the option when I look at my mail folders. Or do I have to enable some Lab option?

**Comment by RainLoop** [comment]
> Did this get implemented?

Yes, you will see these options if your folder list has more than 50 folders.

**Comment by ervee** [comment]
Okay I see. Why not always visible? I'd love to disable some of the unread counters for any of my +/- 40 folders. Or perhaps make the amount of folders configurable in a Labs setting for people which have a slow connection from RainLoop to IMAP Server?

**Comment by RainLoop** [comment]
> Or perhaps make the amount of folders configurable in a Labs setting

```ini
[labs]
folders_spec_limit = 50
```
